### ISSUE TITLE
Using "where" in Select error with Prisma Client v5.19.0

### ISSUE THREAD

**Issue by Chomroeun** [prisma/prisma#25137]
### Bug description

```
return this.prisma.student.paginate({
      page,
      limit,
      select: {
        id: true,
        studentClasses: {
          where: {
            class: {
              is: {
                academicPeriod: {
                  is: {
                    fromDate: { lte: new Date() },
                    toDate: { gte: new Date() },
                  },
                },
              },
            },
          },
          select: {
            id: true,
            class: { select: { id: true, nameEn: true, code: true } },
          },
        },
      },
      orderBy: { id: 'desc' },
    });
```

This query statement works with Prisma Client v5.17.0 with postgreSQL but it return error with Prisma client v5.18.0 and v5.19.0 with error message unknown academicPeriodId.

```
model Student {
  id                 Int       @id @default(autoincrement())
  username           String?   @db.VarChar(20)
  studentClasses                     StudentClass[]
  @@map("students")
}

model StudentClass {
  id                     Int      @id @default(autoincrement())
  studentId              Int
  classId                Int
  student              Student               @relation(fields: [studentId], references: [id], onDelete: Cascade)
  class                Class                 @relation("class", fields: [classId], references: [id], onDelete: Cascade)
  @@map("student_classes")
}

model Class {
  id               Int       @id @default(autoincrement())
  academicPeriodId Int
  academicPeriod                     AcademicPeriod                      @relation(fields: [academicPeriodId], references: [id])
  @@map("classes")
}

model AcademicPeriod {
  id                   Int       @id @default(autoincrement())
  fromDate             DateTime? @db.Date
  toDate               DateTime? @db.Date
  @@map("academic_periods")
}
```

### How to reproduce

1. use Prisma client version 5.17.0
2. update Prisma client version 5.18.0

### Expected behavior

```
model Student {
  id                 Int       @id @default(autoincrement())
  username           String?   @db.VarChar(20)
  studentClasses                     StudentClass[]
  @@map("students")
}

model StudentClass {
  id                     Int      @id @default(autoincrement())
  studentId              Int
  classId                Int
  student              Student               @relation(fields: [studentId], references: [id], onDelete: Cascade)
  class                Class                 @relation("class", fields: [classId], references: [id], onDelete: Cascade)
  @@map("student_classes")
}

model Class {
  id               Int       @id @default(autoincrement())
  academicPeriodId Int
  academicPeriod                     AcademicPeriod                      @relation(fields: [academicPeriodId], references: [id])
  @@map("classes")
}

model AcademicPeriod {
  id                   Int       @id @default(autoincrement())
  fromDate             DateTime? @db.Date
  toDate               DateTime? @db.Date
  @@map("academic_periods")
}
```

### Prisma information

```
model Student {
  id                 Int       @id @default(autoincrement())
  username           String?   @db.VarChar(20)
  studentClasses                     StudentClass[]
  @@map("students")
}

model StudentClass {
  id                     Int      @id @default(autoincrement())
  studentId              Int
  classId                Int
  student              Student               @relation(fields: [studentId], references: [id], onDelete: Cascade)
  class                Class                 @relation("class", fields: [classId], references: [id], onDelete: Cascade)
  @@map("student_classes")
}

model Class {
  id               Int       @id @default(autoincrement())
  academicPeriodId Int
  academicPeriod                     AcademicPeriod                      @relation(fields: [academicPeriodId], references: [id])
  @@map("classes")
}

model AcademicPeriod {
  id                   Int       @id @default(autoincrement())
  fromDate             DateTime? @db.Date
  toDate               DateTime? @db.Date
  @@map("academic_periods")
}
```

### Environment & setup

- Prisma Client: v5.19.0
- Database:  PostgreSQL
- Node.js version: 20

### Prisma Version

```
Prisma Client: v5.19.0
```

**Comment by frontendphil** [comment]
We see similar errors for many `where` queries that traverse relations.

**Comment by cipriancaba** [comment]
I can confirm nested where queries are buggered `The column `t4.status` does not exist in the current database.`

Seems like something changed in the way complex queries are generated. It worked fine for me in 5.18

**Comment by cipriancaba** [comment]
This is our query for reference:

```
await prisma.property.findFirstOrThrow({
        where: {
          id: propertyId,
        },
        include: {
          propertyAddress: true,
          bankAccounts: {
            include: {
              bankAccount: true,
            },
            where: {
              state: 'BOOKED',
              bankAccount: {
                status: 'ACTIVE',
              },
            },
          },
```

**Comment by jkomyno** [comment]
Hi @Chomroeun, I couldn't reproduce the exact scenario your provided because that Postgres schema is invalid. Can you please check `prisma validate` locally to confirm that your schema is valid and up to date, and post it here?

For what is worth, this issue is likely a duplicate of https://github.com/prisma/prisma/issues/25103, and should be fixed by https://github.com/prisma/prisma-engines/pull/4994. We plan to have a patch release for this, Prisma 5.19.1, sometime today. Thanks!

**Comment by Chomroeun** [comment]
```
model AcademicPeriod {
  id                   Int       @id @default(autoincrement())
  nameEn               String    @default("") @db.VarChar(100)
  nameKh               String?   @db.VarChar(100)
  nameCn               String?   @db.VarChar(100)
  description          String?   @db.VarChar(255)
  fromDate             DateTime? @db.Date
  toDate               DateTime? @db.Date
  academicPeriodTypeId Int
  subProgramId         Int
  academicYearId       Int

  classes                Class[]

  @@map("academic_periods")
}

model Class {
  id               Int       @id @default(autoincrement())
  mainProgramId    Int
  subProgramId     Int
  academicPeriodId Int

  academicPeriod                     AcademicPeriod                      @relation(fields: [academicPeriodId], references: [id])
  studentClasses                     StudentClass[]                      @relation("class")

  @@map("classes")
}

model StudentClass {
  id                     Int      @id @default(autoincrement())
  studentId              Int
  classId                Int

  class                Class                 @relation("class", fields: [classId], references: [id], onDelete: Cascade)
  student              Student               @relation(fields: [studentId], references: [id], onDelete: Cascade)
  @@map("student_classes")
}

model Student {
  id                 Int       @id @default(autoincrement())
  username           String?   @db.VarChar(20)
  
  studentClasses                     StudentClass[]
  @@map("students")
}

```

1. pnpm update
2. npx prisma db push --schema prisma/postgres/schema.prisma
3. npx prisma generate --generator client --schema prisma/postgres/schema.prisma
4. npx prisma validate --schema ./prisma/postgres/schema.prisma

The schema at prisma/postgres/schema.prisma is valid 🚀

5. cross-env NODE_ENV=mis nest start mis --watch

```
err: {
      "type": "PrismaClientKnownRequestError",
      "message": "\n\u001b[31mInvalid \u001b[1m`prisma.student.findMany()`\u001b[22m invocation:\u001b[39m\n\n\nThe column `t3.academicPeriodId` does not exist in the current database.",
      "stack":
          PrismaClientKnownRequestError: 
          Invalid `prisma.student.findMany()` invocation:
          
          
          The column `t3.academicPeriodId` does not exist in the current database.
              at Ln.handleRequestError (/Users/s.chomroeuns/Documents/WEG/core_api/node_modules/.pnpm/@prisma+client@5.19.0_prisma@5.19.0/node_modules/@prisma/client/runtime/library.js:121:7753)
              at Ln.handleAndLogRequestError (/Users/s.chomroeuns/Documents/WEG/core_api/node_modules/.pnpm/@prisma+client@5.19.0_prisma@5.19.0/node_modules/@prisma/client/runtime/library.js:121:7061)
              at Ln.request (/Users/s.chomroeuns/Documents/WEG/core_api/node_modules/.pnpm/@prisma+client@5.19.0_prisma@5.19.0/node_modules/@prisma/client/runtime/library.js:121:6745)
              at async Array.$allOperations (/Users/s.chomroeuns/Documents/WEG/core_api/dist/apps/mis/main.js:75824:36)
              at async l (/Users/s.chomroeuns/Documents/WEG/core_api/node_modules/.pnpm/@prisma+client@5.19.0_prisma@5.19.0/node_modules/@prisma/client/runtime/library.js:130:9633)
      "name": "PrismaClientKnownRequestError",
      "code": "P2022",
      "clientVersion": "5.19.0",
      "meta": {
        "modelName": "Student",
        "column": "t3.academicPeriodId"
      },
      "traceId": "2"
    }
```

```
SELECT "t1"."id", "t1"."photoPath", "t1"."username", "t1"."oldUsername", "t1"."lastNameEn", "t1"."firstNameEn", "t1"."lastNameKh", "t1"."firstNameKh", "t1"."gender", "t1"."dob", "t1"."familyProfileId", "t1"."isActive", "t1"."status", "Student_studentClasses"."__prisma_data__" AS "studentClasses" FROM "public"."students" AS "t1" LEFT JOIN LATERAL (SELECT COALESCE(JSONB_AGG("__prisma_data__"), '[]') AS "__prisma_data__" FROM (SELECT "t4"."__prisma_data__" FROM (SELECT JSONB_BUILD_OBJECT('id', "t3"."id", 'classId', "t3"."classId", 'class', "StudentClass_class"."__prisma_data__") AS "__prisma_data__", "t3"."classId", "t3"."academicPeriodId", "t3"."fromDate", "t3"."toDate" FROM (SELECT "t2".* FROM "public"."student_classes" AS "t2" WHERE "t1"."id" = "t2"."studentId" /* root select */) AS "t3" LEFT JOIN LATERAL (SELECT JSONB_BUILD_OBJECT('id', "t6"."id", 'nameEn', "t6"."nameEn", 'code', "t6"."code") AS "__prisma_data__" FROM "public"."classes" AS "t6" WHERE "t3"."classId" = "t6"."id" LIMIT $1) AS "StudentClass_class" ON true /* inner select */) AS "t4" LEFT JOIN "public"."classes" AS "j1" ON ("j1"."id") = ("t4"."classId") LEFT JOIN "public"."academic_periods" AS "j2" ON ("j2"."id") = ("j1"."academicPeriodId") WHERE ("j2"."fromDate" <= $2 AND "j2"."toDate" >= $3 AND ("j2"."id" IS NOT NULL) AND ("j1"."id" IS NOT NULL)) /* middle select */) AS "t5" /* outer select */) AS "Student_studentClasses" ON true WHERE ("t1"."schoolBrandId" = $4 AND ("t1"."id") IN (SELECT "t1"."studentId" FROM "public"."student_campuses" AS "t1" WHERE ("t1"."campusId" = $5 AND "t1"."studentId" IS NOT NULL))) ORDER BY "t1"."id" DESC LIMIT $6 OFFSET $7
```

**Comment by Chomroeun** [comment]
@cipriancaba you are right. v5.17.0 & v5.18.0 works for me too.

**Comment by jkomyno** [comment]
Hi, I confirm this was fixed via https://github.com/prisma/prisma-engines/pull/4994.
We're releasing a patch version for this, Prisma 5.19.1 today! Thanks.

**Comment by apolanc** [comment]
This should now be fixed in the `5.19.1` version, closing this now.
